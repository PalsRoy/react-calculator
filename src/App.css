
.App {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
  width: 100%;
}

.Calculator{
  width: 400px;
  height: 600px;
}

.CalculatorRow {
  display: flex;
  width: 100%;
}

.Result {
  height: 4em;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  font-weight: bold;
  font-size: 1.4em;
  background-color: #000000;
  color: #ffffff;
  padding: 0.5em;
  outline: 1px solid #888888;
}

.Button {
  height: 4em;
  display: flex;
  justify-content: center;
  align-items: center;
  font-weight:bold;
  font-size: 1.4em;
  background-color: #e0e1e6;
  color: black;
  flex: 1;
  outline: 1px solid #888888;
}

.Operator {
  background-color: #fe9241;
  color: #ffffff;
}

.ZeroButton {
  flex-grow: 2.2;
}
