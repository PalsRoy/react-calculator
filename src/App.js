import React, { Component } from 'react';
import './App.css';
import ReactCalulator from './components/calculator';

class App extends Component {
  render() {
    return (
      <div>
          <ReactCalulator />
      </div>
    );
  }
}

export default App;
