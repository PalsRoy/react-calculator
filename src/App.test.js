import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import ReactCalulator from './components/calculator';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('calculator renders correctly', () => {
  const calc = renderer
    .create(<ReactCalulator />)
    .toJSON();
  expect(calc).toMatchSnapshot();
});
