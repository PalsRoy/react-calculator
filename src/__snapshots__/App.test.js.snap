// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculator renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="Calculator"
  >
    <div
      className="Result"
    >
      
    </div>
    <div
      className="CalculatorRow"
    >
      <button
        className="Button"
        onClick={[Function]}
      >
        AC
      </button>
      <button
        className="Button"
        onClick={[Function]}
      >
        +/-
      </button>
      <button
        className="Button"
        onClick={[Function]}
        value="%"
      >
        %
      </button>
      <button
        className="Button Operator"
        onClick={[Function]}
        value="/"
      >
        /
      </button>
    </div>
    <div
      className="CalculatorRow"
    >
      <button
        className="Button"
        onClick={[Function]}
        value="7"
      >
        7
      </button>
      <button
        className="Button"
        onClick={[Function]}
        value="8"
      >
        8
      </button>
      <button
        className="Button"
        onClick={[Function]}
        value="9"
      >
        9
      </button>
      <button
        className="Button Operator"
        onClick={[Function]}
        value="*"
      >
        *
      </button>
    </div>
    <div
      className="CalculatorRow"
    >
      <button
        className="Button"
        onClick={[Function]}
        value="4"
      >
        4
      </button>
      <button
        className="Button"
        onClick={[Function]}
        value="5"
      >
        5
      </button>
      <button
        className="Button"
        onClick={[Function]}
        value="6"
      >
        6
      </button>
      <button
        className="Button Operator"
        onClick={[Function]}
        value="-"
      >
        -
      </button>
    </div>
    <div
      className="CalculatorRow"
    >
      <button
        className="Button"
        onClick={[Function]}
        value="1"
      >
        1
      </button>
      <button
        className="Button"
        onClick={[Function]}
        value="2"
      >
        2
      </button>
      <button
        className="Button"
        onClick={[Function]}
        value="3"
      >
        3
      </button>
      <button
        className="Button Operator"
        onClick={[Function]}
        value="+"
      >
        +
      </button>
    </div>
    <div
      className="CalculatorRow"
    >
      <button
        className="Button ZeroButton"
        onClick={[Function]}
        value="0"
      >
        0
      </button>
      <button
        className="Button"
        onClick={[Function]}
        value="."
      >
        .
      </button>
      <button
        className="Button Operator"
        onClick={[Function]}
      >
        =
      </button>
    </div>
  </div>
</div>
`;
